backend/gateway/gateway.go

```diff
@@ -32,28 +31,31 @@ type ComponentFactory struct {
 }
 
 func Run(f *Flags, cf *ComponentFactory, assets http.FileSystem) {
-	// Use a temporary logger to parse the configuration and output.
-	tmpLogger := newTmpLogger().With(zap.String("file", f.ConfigPath))
+	cfg := MustReadOrValidateConfig(f)
 
-	var cfg gatewayv1.Config
-	if err := parseFile(f.ConfigPath, &cfg, f.Template); err != nil {
-		tmpLogger.Fatal("parsing configuration failed", zap.Error(err))
-	}
-	if err := cfg.Validate(); err != nil {
-		tmpLogger.Fatal("validating configuration failed", zap.Error(err))
-	}
-
-	if f.Validate {
-		tmpLogger.Info("configuration validation was successful")
-		os.Exit(0)
+	// Init a real logger to log the filename.
+	logger, err := newLogger(cfg.Gateway.Logger)
+	if err != nil {
+		tmpLogger := newTmpLogger()
+		defer func() {
+			if err := tmpLogger.Sync(); err != nil {
+				panic(err)
+			}
+		}()
+		tmpLogger.Fatal("could not instantiate logger", zap.Error(err))
 	}
+	defer func() {
+		if err := logger.Sync(); err != nil {
+			panic(err)
+		}
+	}()
```

Comment:
Why was the previous `tmpLogger` so much simpler? Looks like a lot of code for one log statement.

Reply:
the tmp logger is only intended for use when the logger can't be configured. before this change the user will always get log output in mixed formats for the config filename log line.

i can simplify this though if i change the interface to RunWithConfig to accept flags so it has the filename for logging purposes.